src/firstboot/firstboot.c

```diff
@@ -244,7 +246,36 @@ static int should_configure(int dir_fd, const char *filename) {
                 return true; /* missing */
         }
 
-        return arg_force; /* exists, but if --force was given we should still configure the file. */
+        if (arg_force)
+                return true; /* exists, but if --force was given we should still configure the file. */
+
+        if (streq(filename, "passwd")) {
+                _cleanup_fclose_ FILE *passwd = NULL;
+
+                r = xfopenat(dir_fd, filename, "re", O_NOFOLLOW, &passwd);
+                if (r < 0 && r != -ENOENT)
+                        return log_error_errno(r, "Failed to open %s: %m", filename);
+
+                struct passwd *i;
+                while ((r = fgetpwent_sane(passwd, &i)) > 0) {
+                        if (!streq(i->pw_name, "root"))
+                                continue;
+
+                        bool locked = streq_ptr(i->pw_passwd, PASSWORD_LOCKED_AND_INVALID);
```

Comment:
so my thinking was actually to use a different, recognizable token here. i.e. we currently have these three:

* PASSWORD_LOCKED_AND_INVALID (`"!*"`)
* PASSWORD_SEE_SHADOW (`"x"`)
* PASSWORD_NONE (`""`)

And I'd add a fourth one defined like this maybe:

* PASSWORD_UNPROVISIONED (`!unprovisioned`)

And then we make sysusers write that out in case there is no pw provisioned. firstboot would then only look for that.

Explanation for the string:

Valid (modern, i.e. encrypted) password strings have to start with `$`. Password strings starting with `!` are considered "locked" (i.e. tools that can lock/unlock an account generally prefix the password with `!` or drop it again). `x` is only valid in '/etc/passwd` and just means "look in `/etc/shadow` instead". An empty password string `` means: permit login without password. By default, if we want to create a completely locked account we write out `!*`. The `*` has no real meaning, all it is relevant for is that it it's neither `$`, nor `x`, nor `!`, nor an empty string, to avoid confusion  with any of the above. We then prefix it with `!` to indicate that the account is locked. That way we can be sure that any dumb tool that removes the lock prefix still ends up with a set but invalid password, which is good.

Now, what I propose is to use `!unprovisioned` as indicator to our tools that so far no password was set. But it also means hat the account is fully locked, i.e. indicated as "locked", and even when unlocked remains invalid. It's pretty much the same as `!*` in that regard. 

Why not use `!*` for this? simply because having a fully locked root account is totally OK in many setups during normal operation, and we should be able to discern this case from the case where we simply had not opportunity to ask for a pw yet.

Reply:
another maybe even safer option might be to use `!*unprovisioned`, so that we both look like a regular fully locked account (because it starts just like one, i.e. `!*`), but also decorates the string with a suffix that makes clear it's an unprovisioned account.